Nnodes/setup.sh

```diff
@@ -69,11 +69,12 @@ do
     qd=qdata_$n
 
     # Generate the node's Enode and key
-    enode=`docker run -u $uid:$gid -v $pwd/$qd:/qdata $image /usr/local/bin/bootnode -genkey /qdata/dd/nodekey -writeaddress`
+    enode=`docker run -u $uid:$gid -v $pwd/$qd:/qdata $image sh -c "/usr/local/bin/bootnode -genkey /qdata/dd/nodekey -writeaddress; cat /qdata/dd/nodekey"`
+    enode=`docker run -u $uid:$gid -v $pwd/$qd:/qdata $image sh -c "/usr/local/bin/bootnode -nodekeyhex $enode -writeaddress"`
```

Comment:
Could you explain this line?

Reply:
@jososaga Sure thing!

The first line (using `-genkey`) is used to generate the key that will be used by the bootnode.

The second line is used to start the bootnode with the key generated before. At start, the bootnode will write its enode.

With the bootnode's enode in hands, we can setup the other nodes in the network to use the bootnode.

If you wanna know more about this process of setting up a bootnode, you can check this documentation (https://github.com/ethereum/go-ethereum/wiki/Setting-up-private-network-or-local-cluster#private-network)

Reply:
Thanks for the explanation!! I still cannot run the example, I got the above permission error when trying to run the setup.sh script:
Fatal: open /qdata/dd/nodekey: permission denied
It's like it's not generating the key or the user executing the command does not have permission to access the folder.

Do you have any suggestion?
thanks in advance,
Joan.

Reply:
@jososaga
For persmission related issues, please run the following command 
sudo 0755 /qdata_*/dd/nodekey 
for all 3 files.

Reply:
Where should I change the permission: in the Docker file or in the setup script when running the container and change the permission for that folder? The permission error is in the container, not in the host machine.

thanks, Joan.

Reply:
thanks! that solved my problem, I had some permission issues in my windows host machine. 
Now when I try to create the containers: docker-compose up -d, they never start, and this is my constellation log file for qdata_1/logs/constellation.log:

`15:40:42 [INFO] Log level is LevelWarn
15:40:42 [WARN] /qdata/keys/tm.key is too accessible. Please run: chmod 0600 qdata/keys/tm.key
constellation-node: Network.Socket.bind: permission denied (Operation not permitted)`

Does it have something to do that I'm running from a Windows host?
These are the constellation logs for node 2 [constellation.log](https://github.com/ConsenSys/quorum-docker-Nnodes/files/2185264/constellation.log) and node 3 [constellation.log](https://github.com/ConsenSys/quorum-docker-Nnodes/files/2185268/constellation.log)

thanks, Joan.